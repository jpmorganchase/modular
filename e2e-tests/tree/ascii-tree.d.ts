declare module 'ascii-tree' {
  export function generate(input: string): string;
}
