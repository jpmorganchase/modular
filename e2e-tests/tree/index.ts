import fs from 'fs';
import path from 'path';
import asciiTree from 'ascii-tree';
import hash from '@emotion/hash';

function times(str: string, length: number) {
  return Array.from({ length }, () => str).join('');
}

const defaultIgnores = ['node_modules', '.git', '.DS_Store'];
const defaultHashIgnores = [
  // adding lockfiles here because it can be different
  // on different runs; since we install the latest versions
  // of some packages when making a repository
  'yarn.lock',
  'package-lock.json',
  // adding package.json files since versions within these get bumped
  'package.json',
];

function tree(
  _path: string,
  level = 1,
  options = {
    ignores: defaultIgnores,
    hashIgnores: defaultHashIgnores,
  },
): string {
  const stat = fs.statSync(_path);
  if (stat.isDirectory()) {
    const children = fs.readdirSync(_path);
    const dirArr = _path.split('/');
    const dir = dirArr[dirArr.length - 1];
    // todo - should these be sorted?
    // todo - handle symlinks, etc
    return `${times('#', level)}${dir}\n${children
      .filter((child: string) => !options.ignores.includes(child))
      .map((child: string) => tree(path.join(_path, child), level + 1, options))
      .join('\n')}`;
  } else {
    return [
      `${times('#', level)}${path.basename(_path)}`,
      options.hashIgnores.includes(path.basename(_path))
        ? undefined
        : `#${hash(fs.readFileSync(_path, 'utf8'))}`,
    ]
      .filter(Boolean)
      .join(' ');
  }
}

export default function generate(_path: string): string {
  return asciiTree.generate(tree(_path));
}
