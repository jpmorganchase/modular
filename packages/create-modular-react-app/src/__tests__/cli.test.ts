import execa from 'execa';
import * as fs from 'fs-extra';
import * as path from 'path';
import * as tmp from 'tmp';
import tree from 'tree-view-for-tests';

describe('Creating a new modular app via the CLI', () => {
  let cwd: string;
  beforeEach(async () => {
    cwd = path.join(tmp.dirSync().name, 'new-modular-app');

    await execa('yarnpkg', ['create-modular-react-app', cwd], {
      cwd: __dirname,
      cleanup: true,
      stderr: process.stderr,
      stdout: process.stdout,
    });
  });

  afterEach(async () => {
    await fs.remove(cwd);
  });

  it('should create the right tree', () => {
    expect(tree(cwd)).toMatchSnapshot();
  });
});
