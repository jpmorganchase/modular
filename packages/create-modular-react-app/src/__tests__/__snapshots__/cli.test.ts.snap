// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Creating a new modular app via the CLI should create the right tree 1`] = `
"new-modular-app
├─ .editorconfig #1p4gvuw
├─ .eslintignore #1ot2bpo
├─ .gitignore #175wbq
├─ .prettierignore #10uqwgj
├─ .vscode
│  ├─ extensions.json #1i4584r
│  ├─ launch.json #1kk1omt
│  └─ settings.json #xes41c
├─ .yarnrc #1orkcoz
├─ README.md #1nksyzj
├─ modular
│  ├─ setupEnvironment.ts #m0s4vb
│  └─ setupTests.ts #bnjknz
├─ package.json
├─ packages
│  ├─ README.md #14bthrh
│  └─ app
│     ├─ package.json
│     ├─ public
│     │  ├─ favicon.ico #6pu3rg
│     │  ├─ index.html #1m6toxd
│     │  ├─ logo192.png #1nez7vk
│     │  ├─ logo512.png #1hwqvcc
│     │  ├─ manifest.json #19gah8o
│     │  └─ robots.txt #1sjb8b3
│     ├─ src
│     │  ├─ App.css #1o0zosm
│     │  ├─ App.tsx #c80ven
│     │  ├─ __tests__
│     │  │  └─ App.test.tsx #16urcos
│     │  ├─ index.css #o7sk21
│     │  ├─ index.tsx #zdn6mw
│     │  ├─ logo.svg #1okqmlj
│     │  └─ react-app-env.d.ts #t4ygcy
│     └─ tsconfig.json #6rw46b
├─ tsconfig.json #1h72lkd
└─ yarn.lock"
`;
